Source/x11/XGServerEvent.m

```diff
@@ -2683,16 +2682,16 @@ - (NSPoint) mouseLocationOnScreen: (int)screen window: (int *)win
         {
           return p;
         }
-      screen_number = XScreenNumberOfScreen(attribs.screen);
-      if (screen >= 0 && screen != screen_number)
+      screen_id = XScreenNumberOfScreen(attribs.screen);
+      if (screen >= 0 && screen != screen_id)
         {
           /* Mouse not on the requred screen, return an invalid point */
           return p;
         }
       height = attribs.height;
     }
   else
-    height = [self boundsForScreen: screen_number].size.height;
+    height = [self boundsForScreen: screen].size.height;
```

Comment:
In XGServerWindow.m you call this method with a monitor_id. Which one is correct?

Reply:
You're right, it should be monitor_id. I've not figured out how we can get monitor_id here. It's in my todo list.

Reply:
I've checked this code - it's correct. mouseLocationOnScreen:window: correctly called by GUI classes with `screen` parameter set to NSScreen's screenNumber.

Reply:
And NSScreens screenNumbers now are monitor_ids.

Reply:
Exactly. And boundsForScreen: waits for monitor_id.